Marlin/stepper.cpp

```diff
@@ -459,6 +474,10 @@ void Stepper::isr() {
       // #if ENABLED(ADVANCE)
       //   e_steps[TOOL_E_INDEX] = 0;
       // #endif
+
+      #if ENABLED(MK2_MULTIPLEXER)
+        select_multiplexed_stepper(current_block->active_extruder);
```

Comment:
>It would be better to do this selection in the planner instead, and preferably above the planner level

I disagree with this statement. The planner and the levels above it insert stepper commands into a queue. The Stepper isr then processes the commands from that queue and generates stepper pulses. If you switch the multiplexer in the planner I'm then start queuing extruder commands, then the newly selected extruder will receive commands intended for the previously selected extruder until the queue is flushed to the point that the tool change was supposed to occur. 

The other reason that I believe it belongs in the Stepper class is because the multiplexer should really be transparent to the higher level functions. 

I think perhaps there's a disconnect in the definition of the multiplexer. I think of the multiplexer as simply the the electronics board that can drive multiple steppers from a single driver. I think you're lumping the Y splitter that multiplexes multiple filament paths into one. Because of the general purpose applications of using the electronic multiplexer in various different extruder configurations, I think these two pieces should be considered separate. 

By separating the functions of the stepper multiplexer and the filament multiplexer, then the use case for switching from block to block becomes more of a reality. It could even be possible to drive multiple steppers with the multiplexer "simultaneously" by switching the multiplexer and sending step pulses to each stepper in each call to the Stepper isr. However, this may only be feasible on 32-bit applications.